plugins/services/src/js/reducers/serviceForm/FormReducers/__tests__/Containers-test.js

```diff
@@ -0,0 +1,898 @@
+const Containers = require('../Containers');
+const Batch = require('../../../../../../../../src/js/structs/Batch');
+const Transaction = require('../../../../../../../../src/js/structs/Transaction');
+const {ADD_ITEM, SET} = require('../../../../../../../../src/js/constants/TransactionTypes');
+
+describe('Containers', function () {
+  describe('#FormReducer', function () {
+    describe('endpoints', function () {
+      describe('Host Mode', function () {
+
+        it('should have one endpoint', function () {
+          let batch = new Batch();
+
+          batch = batch.add(new Transaction(['containers'], 0, ADD_ITEM));
+
+          batch =
+              batch.add(new Transaction([
+                'containers',
+                0,
+                'endpoints'
+              ], 0, ADD_ITEM));
+
+          expect(batch.reduce(Containers.FormReducer.bind({})))
+          .toEqual([
+            {
+              name: 'container-1',
+              resources: {
+                cpus: 0.1,
+                mem: 128
+              },
+              endpoints: [
+                {
+                  automaticPort: true,
+                  hostPort: null,
+                  labels: null,
+                  name: null,
+                  loadBalanced: false,
+                  vip: null,
+                  protocol: {
+                    tcp: true,
+                    udp: false
+                  },
+                  servicePort: null,
+                  containerPort: null
+                }
+              ]
+            }
+          ]);
+        });
+
+        it('should have one endpoint with name', function () {
+          let batch = new Batch();
+
+          batch = batch.add(new Transaction(['containers'], 0, ADD_ITEM));
+
+          batch =
+              batch.add(new Transaction([
```

Comment:
Could we clean up some of these weird indentations and newlines? This is not really nice to look at